src/wp-includes/class-wp-theme-json.php

```diff
@@ -1540,13 +1546,53 @@ public function merge( $incoming ) {
 	}
 
 	/**
-	 * Returns the slugs for all the presets that cannot be overriden
-	 * in the given path. It returns an associative array
+	 * Returns whether a presets should be overriden or not.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @param array      $theme_json The theme.json like structure to inspect.
+	 * @param array      $path Path to inspect.
+	 * @param bool|array $override Data to compute whether to override the preset.
+	 * @return boolean
+	 */
+	private static function should_override_preset( $theme_json, $path, $override ) {
+		if ( is_bool( $override ) ) {
+			return $override;
+		}
+
+		// The relationship between whether to override the defaults
+		// and whether the defaults are enabled is inverse:
+		//
+		// - If defaults are enabled  => theme presets should not be overriden
+		// - If defaults are disabled => theme presets should be overriden
+		//
+		// For example, a theme sets defaultPalette to false,
+		// making the default palette hidden from the user.
+		// In that case, we want all the theme presets to be present,
+		// so they should override the defaults.
```

Comment:
Core's multiline comment standard uses `/*` instead of `//`. See https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#5-2-multi-line-comments

```suggestion
		/*
		 * The relationship between whether to override the defaults
		 * and whether the defaults are enabled is inverse:
		 *
		 * - If defaults are enabled  => theme presets should not be overriden
		 * - If defaults are disabled => theme presets should be overriden
		 *
		 * For example, a theme sets defaultPalette to false,
		 * making the default palette hidden from the user.
		 * In that case, we want all the theme presets to be present,
		 * so they should override the defaults.
		 */
```